project/paths.ts

```diff
@@ -7,30 +7,45 @@ import { ParsedArgs } from "../parse_args"
 
 // TODO: Do sourceTsPath and destGdPath have to be relative?
 
-export class Paths {
-  /** Where the .ts files live, e.g. ./src */
+export class Paths implements PathProvider {
```

Comment:
Is there a reason to extract this out to an interface? I'm a bit leery of interfaces because they make it harder to go to definition :)

Reply:
One advantage imo is a nicer import structure.
As the places accepting that interface would not need to dig into imports.
Also makes it easier to swap out things for tests.
I think LSP does support a find all implementations shortcut?

Reply:
Ergh... I have bad memories of trying to hunt down implementations of interfaces in the past... I'll try it for now and see how it is. :)